### PR TITLE
update wgpu-core to v0.19

### PR DESCRIPTION
**Pull request by rajveermalviya**
https://github.com/gfx-rs/wgpu/releases/tag/v0.19.0

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,76 +9,121 @@ authors = [
...
+
+[workspace.dependencies.naga]
+package = "naga"
+git = "https://github.com/gfx-rs/wgpu"
+rev = "484457d95993b00b91905fae0e539a093423cc28"
+version = "0.19"
+
 [lib]
-crate-type = ["lib", "cdylib", "staticlib"]
+crate-type = ["cdylib", "staticlib"]
```
Comment (rajveermalviya) [inline Cargo.toml:40]: intentional, don't think we want anyone to use `wgpu-native` as a rust dependency.

**Review by almarklein — APPROVED** [review]

**Comment by hmaarrfk** [comment]
Are there some new compilation instructions with these config changes.

I'm seeing a few warnings and a segfault on my OSX M1 builds. Though the segfault may be unrelated to these warnings. It seems we might be getting some other issues at conda-forge
https://github.com/conda-forge/wgpu-native-feedstock/pull/20


```
warning: unexpected `cfg` condition name: `vulkan`
    --> src/conv.rs:1030:11
     |
1030 |     #[cfg(vulkan)]
     |
     = help: consider using a Cargo feature instead
     = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
              [lints.rust]
              unexpected_cfgs = { level = "warn", check-cfg = ['cfg(gles)'] }
     = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(gles)");` to the top of the `build.rs`
     = note: see <[https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html>](https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html%3E) for more information about checking conditional configuration

warning: unexpected `cfg` condition name: `metal`
    --> src/conv.rs:1568:11
     |
1568 |     #[cfg(metal)]
     |           ^^^^^ help: found config with similar value: `feature = "metal"`
     |
     = help: consider using a Cargo feature instead
     = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
              [lints.rust]
              unexpected_cfgs = { level = "warn", check-cfg = ['cfg(metal)'] }
     = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(metal)");` to the top of the `build.rs`
     = note: see <[https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html>](https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html%3E) for more information about checking conditional configuration

warning: unexpected `cfg` condition name: `metal`
    --> src/conv.rs:1628:11
     |
1628 |     #[cfg(metal)]
     |           ^^^^^ help: found config with similar value: `feature = "metal"`
     |
     = help: consider using a Cargo feature instead
     = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
              [lints.rust]
              unexpected_cfgs = { level = "warn", check-cfg = ['cfg(metal)'] }
     = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(metal)");` to the top of the `build.rs`
     = note: see <[https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html>](https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html%3E) for more information about checking conditional configuration

warning: unexpected `cfg` condition name: `metal`
    --> src/lib.rs:2648:15
     |
2648 |         #[cfg(metal)]
     |               ^^^^^ help: found config with similar value: `feature = "metal"`
     |
     = help: consider using a Cargo feature instead
     = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
              [lints.rust]
              unexpected_cfgs = { level = "warn", check-cfg = ['cfg(metal)'] }
     = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(metal)");` to the top of the `build.rs`
     = note: see <[https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html>](https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html%3E) for more information about checking conditional configuration

warning: unused variable: `metal`
    --> src/conv.rs:1578:5
     |
1578 |     metal: Option<&native::WGPUSurfaceDescriptorFromMetalLayer>,
     |     ^^^^^ help: if this is intentional, prefix it with an underscore: `_metal`
     |
     = note: `#[warn(unused_variables)]` on by default

warning: `wgpu-native` (lib) generated 8 warnings
    Finished `release` profile [optimized] target(s) in 2m 11
```

**Comment by hmaarrfk** [comment]
I think you are seeing the same (scary to me) warnings on your CIs as well:
https://github.com/gfx-rs/wgpu-native/actions/runs/10553889644/job/29234895285

**Comment by rajveermalviya** [comment]
Thanks for noticing that, I wasn't seeing those locally because I hadn't updated to the latest rust version (1.80.1).
Created #413 which should address this issue.

**Pull request merged**